include/realizations/config/formulation.hpp

```diff
@@ -51,11 +52,16 @@ namespace realization{
         if(type=="bmi_multi"){
             for(auto& module : tree.get_child("params.modules")){
                 //Create the nested formulations in order of definition
-                nested.push_back(Formulation(module.second));
+                nested.push_back(Formulation(module.second, mpi_rank));
             }
+        #if NGEN_WITH_MPI
```

Comment:
```c++
#if NGEN_WITH_MPI
int mpi_rank;
MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
if (mpi_rank == 0)
#endif
{
   geojson::...
}
```
Just retrieve the rank and use it right here, and don't change the interfaces of everything in the area to pass it down as an argument.

If you do it this way, no change is needed in any other file.

Reply:
When I did the way as above, I am getting the following compile error:
```
/home/shengting.cui/ngen_multi_out/ngen/include/realizations/config/formulation.hpp:59:23: error: ‘MPI_COMM_WORLD’ was not declared in this scope
         MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
                       ^~~~~~~~~~~~~~
/home/shengting.cui/ngen_multi_out/ngen/include/realizations/config/formulation.hpp:59:9: error: ‘MPI_Comm_rank’ was not declared in this scope
         MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
```

```
 57         #if NGEN_WITH_MPI
 58         int mpi_rank;
 59         MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
```

I think it probably needs 
 `MPI_Init(...);`
 before calling `MPI_Comm_rank(...)` locally which we don't want to do.

Reply:
no, it just needs `#include <mpi.h>` at the top of the file, like any other library would. Guard that under `NGEN_WITH_MPI` too, please.